Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs

```diff
@@ -0,0 +1,135 @@
+using System.Collections;
+using System.Text.Json;
+using Amazon.Lambda.TestTool.Models;
+using Amazon.Lambda.TestTool.Services.IO;
+using Microsoft.AspNetCore.Routing.Template;
+
+namespace Amazon.Lambda.TestTool.Services;
+
+/// <inheritdoc cref="IApiGatewayRouteConfigService"/>
+public class ApiGatewayRouteConfigService : IApiGatewayRouteConfigService
+{
+    private readonly ILogger<ApiGatewayRouteConfigService> _logger;
+    private readonly List<ApiGatewayRouteConfig> _routeConfigs = new();
+
+    /// <summary>
+    /// Constructs an instance of <see cref="ApiGatewayRouteConfigService"/>
+    /// which loads and parses environment variables that match a specific prefix.
+    /// </summary>
+    /// <param name="environmentManager">A service to access environment variables.</param>
+    /// <param name="logger">The logger instance for <see cref="ApiGatewayRouteConfigService"/></param>
+    public ApiGatewayRouteConfigService(
+        IEnvironmentManager environmentManager,
+        ILogger<ApiGatewayRouteConfigService> logger)
+    {
+        _logger = logger;
+        
+        logger.LogDebug("Retrieving all environment variables");
+        var environmentVariables = environmentManager.GetEnvironmentVariables();
+
+        logger.LogDebug("Looping over the retrieved environment variables");
+        foreach (DictionaryEntry entry in environmentVariables)
+        {
+            var key = entry.Key.ToString();
+            if (key is null)
+                continue;
+            logger.LogDebug("Environment variables: {VariableName}", key);
+            if (!(key.Equals(Constants.LambdaConfigEnvironmentVariablePrefix) || 
+                key.StartsWith($"{Constants.LambdaConfigEnvironmentVariablePrefix}_")))
+            {
+                logger.LogDebug("Skipping environment variable: {VariableName}", key);
+                continue;
+            }
+
+            var jsonValue = entry.Value?.ToString();
+            logger.LogDebug("Environment variable value: {VariableValue}", jsonValue);
+            if (string.IsNullOrEmpty(jsonValue))
+                continue;
+
+            try
+            {
+                if (jsonValue.StartsWith('['))
```

Comment:
can you double check if this string needs to have whitespace trimmed in the beginning?

```
 [
 //rest of json
]
```

i believe whitespace is technically ignored in json, so if theres a leading whitespace then this would fail 
```

Reply:
Good catch. Fixed.